packages/react/src/ReactElementValidator.js

```diff
@@ -177,57 +180,93 @@ function validateChildKeys(node, parentType) {
   }
 }
 
-/**
- * Given an element, validate that its props follow the propTypes definition,
- * provided by the type.
- *
- * @param {ReactElement} element
- */
-function validatePropTypes(element) {
-  const type = element.type;
-  let name, propTypes;
+function validatePropTypesRecursively(type, props) {
+  if (type === null || type === undefined || typeof type === 'string') {
+    return;
+  }
+  let propTypes;
+  let innerType;
   if (typeof type === 'function') {
-    // Class or function component
-    name = type.displayName || type.name;
-    propTypes = type.propTypes;
-  } else if (
-    typeof type === 'object' &&
-    type !== null &&
-    type.$$typeof === REACT_FORWARD_REF_TYPE
-  ) {
-    // ForwardRef
-    const functionName = type.render.displayName || type.render.name || '';
-    name =
-      type.displayName ||
-      (functionName !== '' ? `ForwardRef(${functionName})` : 'ForwardRef');
     propTypes = type.propTypes;
-  } else {
-    return;
+  } else if (typeof type === 'object') {
+    switch (type.$$typeof) {
+      case REACT_FORWARD_REF_TYPE: {
+        propTypes = type.propTypes;
```

Comment:
do we warn against propTypes on the inner fn?

Reply:
yes